webapp/packages/plugin-connections/src/ConnectionForm/DriverProperties/ConnectionDriverPropertiesTabService.ts

```diff
@@ -80,6 +80,26 @@ export class ConnectionDriverPropertiesTabService extends Bootstrap {
     const config = contexts.getContext(connectionConfigContext);
 
     config.properties = { ...state.config.properties };
+
+    if (config.driverId) {
+      const driver = this.dbDriverResource.get(config.driverId);
+
+      const driverMap = driver?.driverProperties?.reduce((acc, item) => {
+        acc.set(item.id, item.defaultValue);
+        return acc;
+      }, new Map());
```

Comment:
`const defaultDriverProperties = new Set(driver?.driverProperties?.map(property => property.id) ?? []);`

Reply:
not agreed here. if we use Set, it means that search in this case gonna be O(M). Which leads to for in for cycle which is O(N * M) time complexity. The Map solves this problem due to the fact the search in map is O(1)